commons-mongo/src/main/scala/com/avsystem/commons/mongo/core/ops/BaseFiltering.scala

```diff
@@ -0,0 +1,95 @@
+package com.avsystem.commons
+package mongo.core.ops
+
+import java.util.regex.Pattern
+
+import com.google.common.collect.ImmutableList
+import com.mongodb.client.model.Filters
+import com.mongodb.client.model.geojson.{Geometry, Point}
+import org.bson.BsonType
+import org.bson.conversions.Bson
+import org.mongodb.scala.bson.BsonValue
+
+import scala.util.matching.Regex
+
+trait BaseFiltering[T] extends Any {
+  private[core] def key: String
+  private[core] def encode(t: T): BsonValue
+
+  private def use(t: T)(f: (String, BsonValue) => Bson): Bson = {
```

Comment:
flipping those arguments would be nice for if someone wanted to use point free style xDDD

Reply:
Go away.

Reply:
This argument is point free.